### DISCUSSION TITLE
Filter cleared unexpectedly when adding incompatible criteria

### DISCUSSION THREAD

**Discussion by kiarashi** [PrestaShop/PrestaShop#36365] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When using the Faceted Search module in PrestaShop, I encountered unexpected behavior when selecting filters. Specifically, when selecting both the "Screen Size: 16 inch" and "Screen Size: 17 inch" filters, followed by selecting the "Graphics Card: RTX4050", which is only available in 17-inch (17.3)  laptops, the "Screen Size: 16 inch" filter is automatically cleared. This behavior is not desired as I expect both screen size filters to remain active even after selecting the graphics card filter.

url: https://mkshop.nl/12-laptops

### Expected behavior

After selecting the 'Graphics Card: RTX4050' filter, I expect both the 'Screen Size: 16 inch' and 'Screen Size: 17 inch' filters to remain active, displaying all products with 16-inch and 17-inch screens, in addition to the selected product with the graphics card.

### Steps to reproduce

1. Navigate to the product page with laptops in the "Laptops" category.
2. Select the "Screen Size: 16 inch" filter and "Screen Size: 17 inch" filter.
3. Then, select the "Graphics Card: RTX4050".
4. Observe that the "Screen Size: 16 inch" filter is automatically cleared.

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

8.1.28

### If your bug is related to a module, specify its name and its version

faceted search v3.15.1

**Comment by Hlavtox** [comment]
@kiarashi The filter is automatically cleared because there is no product found that has `GeForce RTX4050 6GB (15231)` and `Screen Size: 16 inch`.

See if you only select the GPU: https://mkshop.nl/12-laptops?q=videokaart-GeForce+RTX4050+6GB+%2815231%29

Nice website BTW. ;-)

**Comment by AureRita** [comment]
Hi @kiarashi 

Thank you for your report, Currently we can't test issue on website directly, but I think @Hlavtox Find the issue, 

Can you tell us if it correct your issue ? 

Waiting for feedback

**Comment by kiarashi** [comment]
@Hlavtox is right with his answer. 

It seems like the AND logic is being used to filter products. Is it possible to change this to OR logic in the future? That way filters aren't automatically removed when there are no products that match all selected criteria.

This change would allow users to see products that match any of the selected filters, providing a more flexible and user-friendly search experience. For example, if both "Screen Size: 16 inch" and "Screen Size: 17 inch" are selected, along with the "Graphics Card: RTX4050" filter, users would see all products with either 16-inch or 17-inch screens, including those with the RTX4050 graphics card.

Could this feature be considered for a future update?

**Comment by florine2623** [comment]
@kiarashi I converted your issue to a Discussion.

A PM will take a look and consider it ^^

Thanks!
